src/type/preload.ts

```diff
@@ -280,13 +281,40 @@ export type DefaultStyleId = {
   defaultStyleId: number;
 };
 
-export type MinimumEngineManifest = {
+export type supportedFeaturesItem = {
+  type: string;
+  value: boolean;
+  name: string;
+};
+
+export const supportedFeaturesItem = z.object({
+  type: z.string(),
+  value: z.boolean(),
+  name: z.string(),
+});
+
+export type SupportedFeatures = z.infer<typeof supportedFeaturesItem>;
+
+export type minimumEngineManifest = {
   name: string;
   uuid: string;
   command: string;
-  port: string;
+  port: number;
+  supported_features: Record<string, supportedFeaturesItem>;
 };
 
+export const minimumEngineManifest = z
```

Comment:
ここもSchemaだと統一感あるかもです
```suggestion
export const minimumEngineManifestSchema = z
```

Reply:
あ、ここ忘れられてるかもです！